[Azure Search][Preview] Stabilize state of feature/search/preview branch

arv100kri opened the pull request:
`feature/search/preview` is going to be the (temporary) ship vehicle for an eventual preview SDK for Azure search (tentatively 11.0.0-preview)

This commit does the following:

1. Regenerates code from swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/7615 that introduced a "base swagger"
2. Re-adds features that were in `2017-11-11-Preview` (and will be in `2019-05-06-Preview`) but not in `2019-05-06`. The only notable change is customer managed encrypted indexes/synonym maps

This is *not* the final version of the preview SDK - we intend to add features (via the swagger spec, to keep it in sync with the REST API). We intend to remove some deprecated types as well.

This PR, sets us up to allow that to happen.

(Optionally) after this PR is checked in, the branch `search-preview` can be eliminated.

Commit:
Generate Microsoft.Azure.Search.Data

Commit:
Generate from preview swagger for Microsoft.Azure.Search.Service

Commit:
Add back encryption tests

Commit:
Fix the test to pass

Commit:
Add session records for encryption tests

Commit:
Update build props and release notes

Commit:
Regenerate the Data directory

Commit:
Regenerate the service directory from master

bleroy commented on sdk/search/Microsoft.Azure.Search/tests/SessionRecords/EncryptionWithCustomerManagedKeysTests/CreateIndexWithCustomerManagedKeysUsingAccessCredentials.json:
```diff
@@ -7,48 +7,48 @@
-          "d9f26adf-bdc6-4251-a0ee-cf3e57bfdb7d"
+          "458eff35-1b96-411c-adb3-f112e2ef3dfc"
         ],
-        "accept-language": [
```
Any particular reason for the casing to change here?

arv100kri replied:
None, but I think it doesn't matter for http headers... This is auto-generated by the test tool, so I didn't hand edit it

bleroy reviewed (approved):
LGTM

Commit:
Update release notes and csproj with correct package version for preview

arv100kri commented:
Need to add session records for all tests, not just the tests that I added.
Will do that and push another commit

Commit:
Update version in ContinuationTokenConverter

Commit:
Update session records to the new api version

arv100kri commented:
> Need to add session records for all tests, not just the tests that I added.
> Will do that and push another commit

Fixed with a recent commit. Will check-in if the tests pass

Commit:
Add one missed out session record
